pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -353,13 +389,17 @@ func InvalidateCache() {
 	configCache = make(map[string]*V2RegistriesConf)
 }
 
+// cacheKey returns the key for the cache which is a concatenated string of the
+// path to registry.conf and the directory. We can do that as both are always
+// non-empty avoiding key colissions.
+func cacheKey(ctx *types.SystemContext) string {
+	return ConfigPath(ctx) + configDirPath(ctx)
```

Comment:
`map` keys can be `struct`s, so this can be precise without having to worry about collisions or ambiguities.

Reply:
Still outstanding.

Reply:
(Using a `struct` would also avoid redundant calls to `ConfigPath` / `configDirPath`; the caller could call “compute paths“ and use the return value for both the cache and accessing the files.)